From ef50d47a314e6906ed2b6e39a5517f6c5937213b Mon Sep 17 00:00:00 2001
From: maxice8 <thinkabit.ukim@gmail.com>
Date: Tue, 18 Sep 2018 17:10:31 -0300
Subject: [PATCH] Makefile: Fix installation and stripping

- Separate stripping into another step of the package
- Conditionalize stripping the binary, distros don't like when packages do
it because they want the debug symbols included and do their own
stripping after extracting them
- use -Dm755 to create the directory (-D) and set the correct executable
mode
- respect PREFIX which can be set to /usr /usr/local and even /
- respect DESTDIR which is set by distros when they are packaging
---
 Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- Makefile
+++ Makefile
@@ -1,8 +1,11 @@
 CC?=cc
 CFLAGS?=-std=c99 -pedantic -Wall -Os
+STRIP?=strip
+PREFIX?=/usr
 
 all:
 	${CC} ${CFLAGS} -o find-cursor find-cursor.c -lX11 -lXext -lXfixes
 
 install:
-	install --strip -o root -g root find-cursor /bin/
+	$(STRIP) find-cursor
+	install -Dm755 find-cursor $(DESTDIR)$(PREFIX)/bin/find-cursor
-- 
2.19.0


